src/Microsoft.ML.TimeSeries/Forecast.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Runtime;
+
+namespace Microsoft.ML.TimeSeries
+{
+    /// <summary>
+    /// Interface for forecasting models.
+    /// </summary>
+    /// <typeparam name="T">The type of values that are forecasted.</typeparam>
+    public interface ICanForecast<out T>
+    {
+        /// <summary>
+        /// Train a forecasting model from an <see cref="IDataView"/>.
+        /// </summary>
+        /// <param name="dataView">Reference to the <see cref="IDataView"/></param>
+        /// <param name="inputColumnName">Name of the input column to train the forecasing model.</param>
+        void Train(IDataView dataView, string inputColumnName);
+
+        /// <summary>
+        /// Update a forecasting model with the new observations in the form of an <see cref="IDataView"/>.
+        /// </summary>
+        /// <param name="dataView">Reference to the observations as an <see cref="IDataView"/></param>
+        /// <param name="inputColumnName">Name of the input column to update from.</param>
+        void Update(IDataView dataView, string inputColumnName);
+
+        /// <summary>
+        /// Perform forecasting until a particular <paramref name="horizon"/>.
+        /// </summary>
+        /// <param name="horizon">Number of values to forecast.</param>
+        /// <returns></returns>
+        T[] Forecast(int horizon);
```

Comment:
>T [](start = 8, length = 1)

Would it make sense to return an `IEnumerable` instead of an array?

Reply:
I am not sure about this, the more presize structure, the easier for the caller to work with it. I think either array or list is better than IEnumarable

---
In reply to: [289154085](https://github.com/dotnet/machinelearning/pull/1900#discussion_r289154085) [](ancestors = 289154085)

Reply:
can you revert this back to array pls or list

---
In reply to: [289501097](https://github.com/dotnet/machinelearning/pull/1900#discussion_r289501097) [](ancestors = 289501097,289154085)

Reply:
Maybe it should take an array as input? Not sure about having a method that allocates an array.

---
In reply to: [289554007](https://github.com/dotnet/machinelearning/pull/1900#discussion_r289554007) [](ancestors = 289554007,289501097,289154085)